### PR TITLE
Login consistency

### PR DESCRIPTION
**Pull request by yhaspel**
Initial development for the login consistency includes: 
X button on relevant Auth flow pages - will take user to Local Node Page in Main.
Completing a flow will take user to Wallet.
Unlocking a wallet takes user to wallet.

### CODE REVIEW

File: app/basicComponents/SmInput.js [hunk 1]
```diff
@@ -28,11 +35,25 @@ const Input = styled.input`
   outline: none;
 `;
 
+const ErrorsSection = styled.div`
+  position: relative;
+  height: 20px;
+`;
+
+const ErrorMessageWrapper = styled.div`
```
Comment (IlyaVi) [inline app/basicComponents/SmInput.js]: combine this and next one into single element

File: app/basicComponents/SmInput.js [hunk 2]
```diff
@@ -41,8 +62,19 @@ class SmInput extends PureComponent<Props> {
   debounce: any;
 
   render() {
-    const { disabled, placeholder, hasError } = this.props;
-    return <Input hasError={hasError} disabled={!!disabled} placeholder={placeholder || INPUT_PLACEHOLDER} onChange={this.onChange} />;
+    const { disabled, placeholder, errorMessage } = this.props;
+    return (
+      <Wrapper>
+        <Input key="input-elem" hasError={errorMessage} disabled={!!disabled} placeholder={placeholder || INPUT_PLACEHOLDER} onChange={this.onChange} />
```
Comment (IlyaVi) [inline app/basicComponents/SmInput.js]: remove key param as it's unused anymore

File: app/screens/main/Main.js [hunk 3]
```diff
@@ -88,13 +87,18 @@ class Main extends Component<Props, State> {
...
               <Route key={route.path} path={route.path} component={route.component} />
             ))}
-            <Redirect to="/main/wallet" />
           </Switch>
         </InnerWrapper>
       </Wrapper>
     );
   }
 
+  componentDidMount() {
```
Comment (IlyaVi) [inline app/screens/main/Main.js]: move this to constructor

File: app/components/auth/StepsContainer.js [hunk 4]
```diff
@@ -45,14 +57,32 @@ type Props = {
...
+  state = {
+    redirectToLocalNode: false,
+    allowClose: true
+  };
+
   render() {
     const { mode } = this.props;
+    const { redirectToLocalNode, allowClose } = this.state;
     const header = mode === authModes.UNLOCK ? 'Welcome Back' : 'Welcome to Spacemesh';
+    if (redirectToLocalNode) {
```
Comment (IlyaVi) [inline app/components/auth/StepsContainer.js]: this is not supposed be to here, redirection decision should be made somewhere else and sure not in render func

File: app/components/auth/StepsContainer.js [hunk 5]
```diff
@@ -45,14 +57,32 @@ type Props = {
...
-class StepsContainer extends PureComponent<Props> {
+type State = {
+  redirectToLocalNode: boolean,
+  allowClose: boolean
+};
+
+class StepsContainer extends PureComponent<Props, State> {
+  state = {
+    redirectToLocalNode: false,
+    allowClose: true
```
Comment (IlyaVi) [inline app/components/auth/StepsContainer.js]: rename to isCloseBtnVisible or something like this with same meaning

File: app/components/auth/StepsContainer.js [hunk 6]
```diff
@@ -45,14 +57,32 @@ type Props = {
   navigateToWallet: Function
 };
 
-class StepsContainer extends PureComponent<Props> {
+type State = {
+  redirectToLocalNode: boolean,
+  allowClose: boolean
+};
+
+class StepsContainer extends PureComponent<Props, State> {
```
Comment (IlyaVi) [inline app/components/auth/StepsContainer.js]: this is not Pure anymore - state is used here

File: app/components/auth/StepsContainer.js [hunk 7]
```diff
@@ -68,11 +98,19 @@ class StepsContainer extends PureComponent<Props> {
...
       default:
         return null;
     }
   };
+
+  handleClose = () => {
+    this.setState({ redirectToLocalNode: true });
+  };
+
+  handleSubModeChange = (subMode: 1 | 2) => {
```
Comment (IlyaVi) [inline app/components/auth/StepsContainer.js]: change to more meaningful name

File: app/screens/main/Main.js (hunk 3 above)
Comment (yhaspel) [reply app/screens/main/Main.js]: This does not achieve the same effect. In order to set the index properly, the component needs to be already mounted.

Comment (IlyaVi) [reply app/screens/main/Main.js]: You need to use location object that comes with component's props and not location on window object that might be not reflect the last changes yet

Comment (yhaspel) [reply app/screens/main/Main.js]: Got it.

**Pull request merged**